certgen/certProcessor/src/main/java/org/incredible/CertificateGenerator.java

```diff
@@ -33,24 +36,34 @@
     private Map<String, String> properties;
     private ObjectMapper objectMapper = new ObjectMapper();
 
-    String directory = "conf/certificate/";
-
     public CertificateGenerator(Map<String, String> properties) {
         this.properties = properties;
     }
 
     private CertificateFactory certificateFactory = new CertificateFactory();
 
 
-    public String createCertificate(CertModel certModel, HTMLTemplateProvider htmlTemplateProvider, String signatureConfig) throws InvalidDateFormatException, WriterException, FontFormatException, NotFoundException, IOException {
+    public String createCertificate(CertModel certModel, HTMLTemplateProvider htmlTemplateProvider, String signatureConfig)
+            throws InvalidDateFormatException, WriterException, FontFormatException, NotFoundException, IOException,
+            SignatureException.UnreachableException, SignatureException.CreationException {
+
         CertificateExtension certificateExtension = certificateFactory.createCertificate(certModel, properties, signatureConfig);
-        generateCertificateJson(certificateExtension);
-        generateQRCodeForCertificate(certificateExtension);
-        if (htmlTemplateProvider.checkHtmlTemplateIsValid(htmlTemplateProvider.getTemplateContent())) {
-            HTMLGenerator htmlGenerator = new HTMLGenerator(htmlTemplateProvider.getTemplateContent());
-            htmlGenerator.generate(certificateExtension);
+        String directory = "conf/" + properties.get(JsonKey.ROOT_ORG_ID).concat("_") + properties.get(JsonKey.TAG).concat("_")
```

Comment:
this logic must not be inside cert processor. Please check the directory, create and delete outside.
All this function will do is to look for a html in a particular structure (orgId, tag which is batchId).